tv_apps/browser/js/sync/db.js

```diff
@@ -194,7 +195,29 @@ var SyncBrowserDB = {
     } else if (indexes.bmkUri) {
       this.db.getBookmarkByUri(indexes.bmkUri, callback);
     } else if (indexes.parentid) {
-      this.db.getBookmarkByParentId(indexes.parentid, callback);
+      this.db.getBookmark(indexes.parentid, folder => {
+        if (folder && folder.type === 'folder' && folder.children) {
+          this.db.getBookmarkByParentId(indexes.parentid, list => {
+            var children = folder.children, result = [];
+            for (var i = 0; i < children.length; i++) {
+              for (var j = 0; j < list.length; j++) {
+                var oriId = list[j].id, id = oriId;
+                // Append the reduntant underline to match the ID from Syncto.
+                if (oriId.length < FXSYNCID_LEN) {
+                  id += new Array(FXSYNCID_LEN + 1 - oriId.length).join('_');
```

Comment:
Can this pass jshint? ```new Array``` is not recommanded.

An additional thought is, according to http://stackoverflow.com/a/14343876, the ```new Array``` padding method is very slow (though probably not much difference in this case). Please use the method described in http://stackoverflow.com/a/24398129 .